### PR TITLE
feature: add selection of config file and run on single cluster

### PR DESCRIPTION
**Pull request by blaisedias**
Add options to
- select the config file for the test run
	- selfci_config.yaml
	- hcloudci_config.yaml (default)
- run all tests on a single cluster

These changes are precursor to fixing the CI to run,
using generic-system-test

Which will be required when we move the jenkins scripts
out of the mayastor-e2e repo.

### CODE REVIEW

File: pipelines/common2/common.groovy [hunk 1]
```diff
@@ -423,8 +423,17 @@ def RunOneTestPerCluster(e2e_test, loki_run_id, params) {
         mkdir -p "${reports_dir}"
         nix-shell --run './scripts/get_cluster_env.py --platform "${test_platform}" --oxray  "${envs_txt_file}" --oyaml "${envs_yaml_file}"'
     """
-    
-    def cmd = "cd ${e2e_dir} && ./scripts/e2e-test.sh --device /dev/sdb --tag \"${e2e_image_tag}\"  --onfail stop --tests \"${testset}\" --loki_run_id \"${loki_run_id}\" --loki_test_label \"${e2e_test}\" --reportsdir \"${env.WORKSPACE}/${e2e_reports_dir}\" --registry \"${env.REGISTRY}\" --session \"${session_id}\" --ssh_identity \"${env.WORKSPACE}/${e2e_environment}/id_rsa\" --product \"${product}\" "
+
+    // compose optional command line options
+    def opts = ''
```
Comment (chriswldenyer) [inline pipelines/common2/common.groovy:428]: I suggest calling this "config_file_opt" so it is obvious what it represents on line 436

File: pipelines/common2/common.groovy [hunk 2]
```diff
@@ -423,8 +423,17 @@ def RunOneTestPerCluster(e2e_test, loki_run_id, params) {
...
+    try {
+        if (params.e2e_config_file != null && params.e2e_config_file != '') {
+            opts = "--config \"${params.e2e_config_file}\" "
+        }
+    } catch (err) {
```
Comment (chriswldenyer) [inline pipelines/common2/common.groovy:433]: This could do with a comment explaining the need for try-catch here

**Review by chriswldenyer — APPROVED** [review]
LGTM, some suggestions

File: pipelines/common2/common.groovy (hunk 2 above)
Comment (blaisedias) [reply pipelines/common2/common.groovy:433]: I let my pythonic instincts get the better of me here. 
Changed to use containsKey.

File: pipelines/common2/common.groovy (hunk 1 above)
Comment (blaisedias) [reply pipelines/common2/common.groovy:430]: https://github.com/mayadata-io/mayastor-e2e/pull/477/commits/aadf28bbb51737ecadbd67cc0f0e95695b0e4721

File: pipelines/common2/common.groovy (hunk 2 above)
Comment (blaisedias) [reply pipelines/common2/common.groovy]: https://github.com/mayadata-io/mayastor-e2e/pull/477/commits/aadf28bbb51737ecadbd67cc0f0e95695b0e4721

**Review by rohan2794 — APPROVED** [review]
LGTM

**Pull request merged**